pages/pages/[userOrg]/[filterName]/[toolName].tsx

```diff
@@ -58,10 +88,31 @@ export const getServerSideProps = async (ctx: GetServerSidePropsContext) => {
     };
   }
 
+  // Keeping this here so we are sure the page is not private before we fetch the social card.
+  async function fetchSocialCardURL() {
```

Comment:
Turn this into a reusable utility function and put it in `lib/utils`. The only thing that changes is the API path for the user/highlight/insight. We can refactor those in a separate PR.

Example code:

```ts
const ogImage = await fetchSocialCardURL(`/insights/${insightId}`); // returns image URL or empty string
```